x/superfluid/keeper/stake.go

```diff
@@ -287,6 +287,48 @@ func (k Keeper) undelegateCommon(ctx sdk.Context, sender string, lockID uint64)
 	return intermediaryAcc, nil
 }
 
+// partialSuperfluidUndelegate acts similarly to undelegateCommon, but undelegates a partial amount of the lock's delegation rather than the full amount. The amount
+// that is undelegated is placed in a new lock. This function returns the intermediary account associated with the original lock ID as well as the new lock that was created.
+// An error is returned if the amount to undelegate is greater than the locked amount.
+func (k Keeper) partialSuperfluidUndelegate(ctx sdk.Context, sender string, lockID uint64, amountToUndelegate sdk.Coin) (types.SuperfluidIntermediaryAccount, *lockuptypes.PeriodLock, error) {
```

Comment:
consider returning named arguments to avoid confusion

Reply:
Done here https://github.com/osmosis-labs/osmosis/pull/5162/commits/0d1e2c59dc33aa6b49fb8f20d699fe28ce8d8148